src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -580,6 +586,129 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
     }
   }
 
+  public connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener("paste", this._handlePaste);
+  }
+
+  public disconnectedCallback() {
+    window.removeEventListener("paste", this._handlePaste);
+    super.disconnectedCallback();
+  }
+
+  private _handlePaste = async (ev: ClipboardEvent) => {
+    // Ignore events on inputs/textareas
+    const target = ev.composedPath()[0];
+    if (
+      target instanceof HTMLElement &&
+      (target.tagName === "INPUT" || target.tagName === "TEXTAREA")
+    ) {
+      return;
+    }
+
+    const paste = ev.clipboardData?.getData("text");
+    if (!paste) {
+      return;
+    }
+
+    const loaded: any = load(paste);
+    if (loaded) {
+      let normalized: AutomationConfig | undefined;
+
+      try {
+        normalized = normalizeAutomationConfig(loaded);
```

Comment:
I think we should create a struct for automations, with which we validate the pasted config